src/ejs/uptini.1

```diff
@@ -0,0 +1,41 @@
+title UPTINI - Create UPTIME DATA
+
+chdsko=10
+
+flose=.lose %lsfil
+
+data:
+	IRP M,,[DB,NO,UP,ES,LX]
+        sixbit/M/		; sixbit host name
+	0			; sixbit date
+	0			; sixbit time
+	TERMIN
+	0	;; end of data
+dataln==.-data
+
+datptr: -dataln,,data
+
+go:	.call [ setz
+		sixbit /open/
+		[.bio,,chdsko]
+		dbdev
+		dbfn1
+		dbfn2
+		setz dbdir ]
+	 flose
+	.call [ setz
+		sixbit /iot/
+	        [%climm,,chdsko]
```

Comment:
This doesn't work as intended.  Either `%climm,,chdsko` for an immediate argument, or `[chdsko]`.

Reply:
Then why does the code work? I think I tried one of the two alternatives you suggested and it didn’t work. Running my code does write the file. 

Why do you think it is wrong? Sure you’re  it thinking of .iot uuo?

Reply:
I did find this email:

```
Date: 14 January 1983 02:57-EST
From: Alan Bawden <ALAN @ MIT-MC>
Subject:  IOT control bits
To: BUG-ITS @ MIT-MC
cc: GSB @ MIT-MC

This doesn't work as advertized:

	.call [setz ? sixbit /iot/
		[%tipek,,ttyich]
		setzm t]

While this does:

	.call [setz ? sixbit /iot/
		%clbit,,%tipek
		movei ttyich
		setzm t]

We could just fix the documentation of the IOT call to not claim that the
left half of the first argument is XORed with the control bits...
```

which tends to support what you're saying. But when I tested my code, it worked fine.

Reply:
Since it is slightly more efficient, I changed the [%climm,,chdsko] to %climm,,chdsko, and it still works.  Since this was one of your suggested alternatives, and since I see *NO* code in ITS that does it my way or, for that matter, the %climm,,chdsko way -- everyone does a MOVEI CHDSKO, I'm happy to change it to not use an indirect reference.  But, as I said above, it appears to work fine.

Reply:
Sure, `[%climm,,chdsko]` works, just not what you (probably) intended.  From the use of %climm, it looks like you intended an immediate argument.  But to do that you'll need to remove the brackets, because the %climm must be in the argument block.  It works with the bracked because then it's a %clin argument instead and then I suppose ITS takes the channel number from the right half or maybe even just the low 4 bits.

Reply:
Ok. I did intend an immediate argument. I understand (now) your point. I did update the code to not indirect. What I was confused about was that the code worked fine. Thanks for your %clin explanation — i’ll Read up on it, since i’ve forgotten these details with the long passage of time.

Reply:
Of course.  Reading source code doesn't make it more clear, since you can see `foo` or `%clin,,foo` or `move foo` and they all work the same.  Likewise `%climm bar` or `movei bar`.  And then there are everyone's macro packages on top of that.

Reply:
Also, %CLINN,,FOO is the same as FOO, but MOVE FOO isn't (quite) the same, since MOVE is 200000, and therefore MOVE FOO is 200000,FOO.  Maybe the 2000000 bit will be ignored.  In any case, MOVE obscures what is really intended here.

Reply:
To get the correct bits for %CLIMM,,FOO, which is what I wanted, I would have had to use MOVEI FOO, which would have a) included the 200000 bit (unwanted) and b) obscured the intent.  MOVE FOO doesn't signify "immediate".

Reply:
Beating a dead horse here I suppose, but...  Note that the %CL bits agree with the semantics of the MOVE instructions.  %CLIMM - MOVEI (immediate), %CLIN - MOVE (from memory), %CLOUT - MOVEM (to memory).  I think it's mnemonic.

Reply:
Sure, but we're not "moving" anything, nor executing any PDP-10 instructions.  We're specify a parameter to a system call.  Obviously, some clever ITS people picked the bits for signifying memory and immediate cleverly.  But clever people aren't always people who are making things easier for others to understand.  %CLIMM,,FOO is the *right* way of doing this.  MOVEI FOO works, but is oblique.